orangecontrib/text/corpus.py

```diff
@@ -88,6 +90,14 @@ def _infer_text_features(self):
         self.set_text_features(include_feats)
 
     def extend_corpus(self, metadata, Y):
```

Comment:
Could be. However, this is kind of tailored for NYT where `Y` is always present. To make it more general, we should also consider `X`. I suggest this to be done in another PR.

About `Table.concatenate(..., axis=1)`, maybe `extend` would be more appropriate when it is fixed 😄